Fix hard crash when USERNAME is not set as env variable on Windows

jellespijker opened the pull request:
Changelog: Bugfix: Catching `KeyError` if `USERNAME` is not set as env variable on Windows.
Docs: omit
Fixes #11222

using `os.environ` fails on __getitem__ when environmental variable
is not set. ~~Using getenv~~ catching the KeyError ensures it fails
gracefully ensuring that no hard crash occurs.

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [ ] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

jellespijker committed:
use getenv to obtain USERNAME

using `os.environ` fails on __getitem__ when environmental variable
is not set. Using getenv fails gracefully making sure no hard crash
occurs.

Fixes #11222

memsharded commented on conans/util/windows.py:
```diff
@@ -85,7 +85,7 @@ def path_shortener(path, short_paths):
     # access problems in cygwin/msys2 windows subsystems when using short_home folder
     try:
-        userdomain, username = os.getenv("USERDOMAIN"), os.environ["USERNAME"]
+        userdomain, username = os.getenv("USERDOMAIN"), os.getenv("USERNAME")
```
I guess it is not an issue that the domain is called ``None\None``, is it?
If the ``cacls`` command is not doing anything with this domain, then maybe we could just capture the ``KeyError`` in the exception and fail faster?

jellespijker replied:
Failing faster always has my preference. I can revert this commit and catch the `KeyError` 

or we can accept that  the used domain will be `None` in my case  and executing that command `cacls <short_home_path> /E /G "None":F` exits with `Invalid arguments` and has no effect when I run it in Powershell.

But both solutions also mean that the workaround for the `short_paths` in cygwin/msys2 subsystems will not work when both `USERDOMAIN` and `USERNAME` are not set as environmental variables. I could attempt to fix the workaround, if I could reproduce the original behavior, but it might be better to do this in a separate issue and PR. A possible fix for the workaround is using the output of `whoami`?

powershell
```powershell
PS C:\dev> whoami
<domain>\<username>
```
cmd
```cmd
C:\dev>whoami
<domain>\<username>
```

mingw is the odd duck and would need some extra processing. Domain is capitalized and the `\` has been replaced with `+`
```shell
MINGW64 /c/dev
$ whoami
<DOMAIN>+<username>
```

Which approach has your preference @memsharded

memsharded replied:
> Failing faster always has my preference. I can revert this commit and catch the KeyError

Yes, lets do ``KeyError`` and fail fast.
Keep this PR just for that, lets solve this, which is easy and low risk.

Regarding the cygwin/msys2 subsystems I am not that concerned. To be honest, the whole ``short_paths`` thing disappear in 2.0, so as long as there is some workaround, I wouldn't put much effort on this.

memsharded reviewed (commented):
Thanks for contributing this fix

jellespijker committed:
Revert "use getenv to obtain USERNAME"

This reverts commit b18a9f2b945e45e9523e5cb9a2e187f234cda47a.

jellespijker committed:
Catch KeyError missing USERNAME env variable

os.environ throws a KeyError when a variable is not
set. Catching that error allows it to fail gracefully.

Fixes #11222
